client/src/util/fetchFromAPI.js

```diff
@@ -18,8 +16,9 @@ export const queryAPI = async (options, toast) => {
   };
   const response = await fetch(url, method);
   if (response.ok) {
-    return (result.data = await response.json());
+    result.data = await response.json();
   } else {
-    toast.error(`Food item not saved.  Error: ${result.error}`);
+    result.error(`Food item not saved.  Error: ${result.error}`);
```

Comment:
Did you mean to change this from `toast` to result`?  It's probably a good time to over this branch of the condition with a test so you can validate that it works as expected.  How do you feel about tackling that?

Reply:
I'm good to give it a try 💯

Reply:
Let's focus our efforts here for a little bit.  Taking the convo to Slack though.

Reply:
<img width="489" alt="Screen Shot 2023-02-14 at 9 44 42 AM" src="https://user-images.githubusercontent.com/34128735/218771981-625e4831-aa70-4100-9c52-f756aaed2051.png">

Pushed new changes - I'm I heading in the right direction? Currently failing with error no options passed

Reply:
You've got a test in place, but it's not testing the thing I had in mind.  I'm clearly doing a terrible job expressing myself 😞 

The test should do the following

- Force `fetch` to receive a non-200 response (from the mock)
- verify that `queryAPI` returns an object with an `error` property
- verify that the `error` property is not null
- verify that the `error` property contains the response message from the server

That first bullet point can be extra tricky.  Have a look at the `jest-fetch-mock` documentation for mocking a failed fetch.  https://github.com/jefflau/jest-fetch-mock#mocking-a-failed-fetch

Reply:
<img width="391" alt="Screen Shot 2023-02-16 at 9 04 06 PM" src="https://user-images.githubusercontent.com/34128735/219530937-6e6874a4-e536-4290-a31a-2d32686c2192.png">

Reply:
Totally forgot to git push ...pushed for review - sorry that was just sitting somewhere 👎